pandas/io/formats/xml.py

```diff
@@ -0,0 +1,763 @@
+        self.frame_dicts = self.process_dataframe()
+        self.handle_indexes()
+        self.prefix_uri = self.get_prefix_uri()
+
+    def build_tree(self) -> bytes:
+        """
+        Build tree from  data.
+
+        This method initializes the root and builds attributes and elements
+        with optional namespaces.
+        """
+        from xml.etree.ElementTree import Element, SubElement, tostring
+
+        self.root = Element(
+            f"{self.prefix_uri}{self.root_name}", attrib=self.other_namespaces()
+        )
+
+        for k, d in self.frame_dicts.items():
+            self.d = d
+            self.elem_row = SubElement(self.root, f"{self.prefix_uri}{self.row_name}")
+
+            if self.attr_cols:
+                self.build_attribs()
+            if self.elem_cols:
+                self.build_elems()
+            if not self.attr_cols and not self.elem_cols:
+                self.elem_cols = list(self.frame_dicts[0].keys())
+                self.build_elems()
+
+        self.out_xml = tostring(self.root, method="xml", encoding=self.encoding)
+
+        if self.pretty_print:
+            self.out_xml = self.prettify_tree()
+
+        if not self.xml_declaration:
+            self.out_xml = self.remove_declaration()
+
+        if self.stylesheet:
+            warn(
+                "To use stylesheet, you need lxml installed. "
+                "The non-transformed, original XML is returned instead.",
+                UserWarning,
+            )
+
+        return self.out_xml
+
+    def validate_columns(self) -> None:
+        """
+        Validate elems_cols and attrs_cols.
+
+        This method will check if columns is list-like.
+
+        Raises
+        ------
+        ValueError
+            * If value is not a list and less then length of nodes.
+        """
+        if self.attr_cols and not is_list_like(self.attr_cols):
+            raise TypeError(
+                f"{type(self.attr_cols).__name__} is not a valid type for attr_cols"
+            )
+
+        if self.elem_cols and not is_list_like(self.elem_cols):
+            raise TypeError(
+                f"{type(self.elem_cols).__name__} is not a valid type for elem_cols"
+            )
+
+    def validate_encoding(self) -> None:
+        """
+        Validate encoding.
+
+        This method will check if encoding is among listed under codecs.
+
+        Raises
+        ------
+        LookupError
+            * If encoding is not available in codecs.
+        """
+
+        try:
+            codecs.lookup(self.encoding)
+        except LookupError as e:
+            raise e
+
+    def process_dataframe(self) -> None:
+        """
+        Adjust Data Frame to fit xml output.
+
+        This method will adjust underlying data frame for xml output,
+        including replacing missing entities and including indexes.
+        """
+
+        na_dict = {"None": self.na_rep, "NaN": self.na_rep, "nan": self.na_rep}
```

Comment:
these are usually called `na_values`. make this a class variable / optionally as a followup make this configurable.

Reply:
can you do this

Reply:
@ParfaitG can you update this